apps/dashr-financial-report/assets/FinancialFunctions.R

```diff
@@ -0,0 +1,130 @@
+# Install a package that is necessary and has not been installed, or load one that is.
+
+usePackage <- function(p) 
+{
+  if (!is.element(p, installed.packages()[,1]))
+    install.packages(p, dep = TRUE)
+  require(p, character.only = TRUE)
+}
+
+
+# Generate a Table Function 
+generate_table <- function(df, nrows=20)
+  
+  # function generates a dash table from a supplied data frame (df)
+  
+  #  and number of rows (nrows) to display
+{
+  
+  n <- min(nrows, nrow(df))
+  
+  rows <- lapply(seq(1, n), function(i) {
+    
+    htmlTr(children = lapply(as.character(df[i,]), htmlTd))
+    
+  })
+  
+  header <- htmlTr(children = lapply(names(df), htmlTh))
+  
+  htmlTable(
+    
+    children = c(list(header), rows)
+  )
+}
+
+# Remove.Factors
+
+remove.factors <- function (df) 

```

Comment:
Not 100% certain, but I suspect you could simplify this, e.g.

```
f2n <- function(x) as.numeric(levels(x))[x]
f2c <- function(x) as.character(levels(x))[x]
```

Reply:
Simplification worked, no notable changes with how it affects the tables, but the code is a lot cleaner.